src/graph.py

```diff
@@ -0,0 +1,178 @@
+"""Graphy mcgraphface"""
+
+class Graph(object):
+    """Define the Link_List class structure"""
+    def __init__(self):
+        """Make an empty dictionary."""
+        self.graph_dict = {}
+
+
+    def add_node(self, value):
+        """Check if node of given value exists in dictionary.
+        If not, add it with an empty list."""
+        try:
+            self.graph_dict[value]
+        except KeyError:
+            self.graph_dict[value] = []
+
+    def add_edge(self, val1, val2, weight=0):
+        """Ensure that nodes of val1 and val2 exist (creating them if they don't.
+        Then make an edge connecting val1 to val2."""
+        self.add_node(val1)
+        self.add_node(val2)
+        self.graph_dict[val1].append([val2, weight])
+
+    def nodes(self):
+        """Return a list of all keys in dictionary."""
+        to_return = []
```

Comment:
There is an inbuilt dictionary method which would allow you to do this one line.